packages/test/service-load-test/src/nodeStressTest.ts

```diff
@@ -180,18 +183,35 @@ async function runnerProcess(
     runId: number,
     url: string,
 ): Promise<number> {
+    const client = applicationInsights.defaultClient;
+
+    client.trackMetric({name: "Test Client Started", value: 1});
+    client.trackTrace({message: `${runId} Starting client with url: ${url}`});
+
     try {
         const runConfig: IRunConfig = {
             runId,
             testConfig: profile,
         };
         const stressTest = await load(loginInfo, url);
+
         await stressTest.run(runConfig);
         console.log(`${runId.toString().padStart(3)}> exit`);
+
+        client.trackTrace({message: `Completed client with url: ${url}`});
+        client.trackMetric({name: "Test Client Successful", value: 1});
+
+        client.flush();
         return 0;
     } catch (e) {
         console.error(`${runId.toString().padStart(3)}> error: loading test`);
         console.error(e);
+
+        client.trackTrace({message: `${runId} Error: ${e}`});
```

Comment:
may we can move this to a separate function, which takes client + three optional arguments, trace, exception and metric.
Atleast one of them should be present. Based on what data is present it does trackX of that and then flushes it.

Reply:
Refactored code for better maintenance and readability.